elastic/kibana
# [Console] Improve error message when localStorage quota is reached

## Summary

Fix https://github.com/elastic/kibana/issues/78014

* Added logic for identifying quota related error messages
* Reduced the number of requests stored in console history from 500 to 100 (not sure having last 500 requests is meaningful?)
* Updated copy displayed on toast when localStorage quota is reached
* Error notifications should only show once, even when sending multiple requests

Using error name to uniquely identify the quota storage exceeded error as listed [here](https://developer.mozilla.org/en-US/docs/Web/API/DOMException). But there might be a better way to do this?

~Still need to add a test.~

### How to test

Fill up your localStorage, e.g.:

```js
for (let x=0;x<100000000;x++) {
  localStorage.setItem(`my_${x}`, 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');
}
```

Send a request that should save to localStorage, the error notification should appear.

Empty localStorage when done:

```js
for (let x=0;x<100000000;x++) {
  localStorage.removeItem(`my_${x}`);
}
```

### Screenshot

<img width="330" alt="Screenshot 2021-03-05 at 16 03 07" src="https://user-images.githubusercontent.com/8155004/110133193-4dfa9400-7dcc-11eb-9e2f-0382f9a0adc9.png">

### Release note

When Console's history fills up we provide a more helpful error message. To help decrease Console's dominance of localStorage we also store fewer requests (last 100 instead of last 500) and ensured that the notification for emptying storage is shown once even when sending multiple requests at once (by selecting multiple requests in the editor).

### Checklist

Delete any items that are not applicable to this PR.
